scripts/smoldyn.py

```diff
@@ -1,57 +1,138 @@
+                    input_data_url=model_data.get("input_data_url", ""),
+                    raw_output_data_url=model_data.get("raw_output_data_url", ""),
                 ),
-                doi="10.1016/j.bpj.2016.02.002",
-                source_code_url="https://github.com/allen-cell-animated/simulariumio",
-                source_code_license_url="https://github.com/allen-cell-animated/simulariumio/blob/main/LICENSE",
-                input_data_url="https://allencell.org/path/to/native/engine/input/files",
-                raw_output_data_url="https://allencell.org/path/to/native/engine/output/files",
-            ),
+            )
+
+    # spatial units defaults to 1.0 meters
+    spatial_units = UnitData(
+        name=event["spatial_units"]["name"]
+            if "spatial_units" in event and "name" in event["spatial_units"]
+            else "meter",
+        magnitude=float(event["spatial_units"]["magnitude"])
+            if "spatial_units" in event and "magnitude" in event["spatial_units"]
+            else 1.0,
+    )
+
+    # time units default to 1.0 seconds
+    time_units = UnitData(
+        name=event["time_units"]["name"]
+            if "time_units" in event and "name" in event["time_units"]
+            else "second",
+        magnitude=float(event["time_units"]["magnitude"])
+            if "time_units" in event and "magnitude" in event["time_units"]
+            else 1.0,
+    )
+
+    data = SmoldynData(
+        meta_data=MetaData(
+            box_size=box_size,
+            trajectory_title=event.get("trajectory_title", "No Title"),
+            scale_factor=float(event.get("scale_factor", 1.0)),
+            camera_defaults=camera_data,
+            model_meta_data=model_meta_data,
         ),
         smoldyn_file=InputFileData(
-            file_contents=event["file_contents"],
+            file_contents=event["file_contents"]["file_contents"],
         ),
-        display_data={
-            "red(solution)": DisplayData(
-                name="B",
-                radius=1.0,
-                display_type=DISPLAY_TYPE.OBJ,
-                url="b.obj",
-                color="#dfdacd",
-            ),
-            "green(solution)": DisplayData(
-                name="A",
-                radius=2.0,
-                color="#0080ff",
-            ),
-        },
-        time_units=UnitData("ns"),  # nanoseconds
-        spatial_units=UnitData("nm"),  # nanometers
+        display_data=display_data,
+        time_units=time_units,
+        spatial_units=spatial_units,
     )
-    
-    c = SmoldynConverter(example_data)
-    translation_magnitude = -box_size / 2
-    filtered_data = c.filter_data([
-        TranslateFilter(
-            translation_per_type={},
-            default_translation=translation_magnitude * np.ones(3)
-        ),
-    ])
-    
-    return TrajectoryConverter._read_trajectory_data(filtered_data)
+
+    return JsonWriter.format_trajectory_data(SmoldynConverter(data)._data)
```

Comment:
it seems like we should not have to access a private element here. At one point I did write a public method that returned the data as JSON but that was before that was factored into a separate writer, but it would be worth checking if it exists still

Reply:
Also I'm getting an error here when I try to submit from the branch you have in your PR. 
```
"  File "/var/task/smoldyn.py", line 138, in lambda_handler
    return JsonWriter.format_trajectory_data(SmoldynConverter(data)._data)
"
```
`"'NoneType' object is not subscriptable", errorType: 'TypeError', stackTrace: Array(3)}`

Unfortunately it's not telling me what thing came in as a NoneType.

Reply:
What have you inputted to get that error message? I'm not getting it when I run that locally, and I wanna figure this out

Reply:
I couldn't get that same error again, but i got this one: 
<img width="915" alt="Screen Shot 2022-10-04 at 12 36 19 PM" src="https://user-images.githubusercontent.com/5170636/193910076-c7d4e753-3740-4497-b2aa-d8df87123acd.png">

Reply:
I was able to reproduce: 
It seems to be caused by the camera position being filled out:

<img width="466" alt="Screen Shot 2022-10-04 at 1 19 07 PM" src="https://user-images.githubusercontent.com/5170636/193918735-aabf8aa7-4380-44ee-b558-a700f67fe4c9.png">

Reply:
Alright, I just pushed a change that fixes that, as well as another bug I found in how I was handling vector positions